sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala

```diff
@@ -3034,12 +3035,28 @@ object GpuOverrides extends Logging {
           if (!conf.isApproxPercentileEnabled) {
             willNotWorkOnGpu(s"approx_percentile on GPU is disabled. " +
               s"Set ${RapidsConf.ENABLE_APPROX_PERCENTILE}=true to enable it")
-          } else if (!childExprs.tail.forall(_.wrapped match {
-            case lit: Literal => lit.value != null
-            case _ => false
-          })) {
-            willNotWorkOnGpu("approx_percentile on GPU only supports non-null literal " +
-              "expressions for percentiles and accuracy")
+          } else {
+
+            // check if the percentile expression can be supported on GPU
+            childExprs(1).wrapped match {
+              case lit: Literal => lit.value match {
+                case null =>
+                  willNotWorkOnGpu(
+                    "approx_percentile on GPU only supports non-null literal percentiles")
+                case a: GenericArrayData if a.array.isEmpty =>
+                  willNotWorkOnGpu(
+                    "approx_percentile on GPU does not support empty percentiles arrays")
+                case a: GenericArrayData if a.array.contains(null) =>
```

Comment:
We ran into issues not that long ago with UnsafeArrayData instead of GenericArrayData in a literal.  That is because the ArrayData was the result of running a foldable method instead of just generating the literal directly.   Would be good to work with just ArrayData and avoid using `.array` for it because UnsafeArrayData does not support `.array`